adalflow/adalflow/components/model_client/lm_studio_client.py

```diff
@@ -0,0 +1,120 @@
+import os
+import re
+from typing import Dict, Optional, Any, List, Union, Sequence
+import logging
+import backoff
+import requests
+from adalflow.core.types import ModelType, GeneratorOutput, EmbedderOutput, Embedding, Usage
+from adalflow.core.model_client import ModelClient
+
+log = logging.getLogger(__name__)
+
+class LMStudioClient(ModelClient):
+    """A component wrapper for the LM Studio API client."""
+
+    def __init__(self, host: Optional[str] = None, port: Optional[int] = None):
+        super().__init__()
+        self._host = host or os.getenv("LMSTUDIO_HOST", "http://localhost")
+        self._port = port or int(os.getenv("LMSTUDIO_PORT", "1234"))
+        self._base_url = f"{self._host}:{self._port}/v1"
+        self.init_sync_client()
+        self.async_client = None  # To be added
+
+    def init_sync_client(self):
+        """Create the synchronous client"""
+        self.sync_client = requests.Session()
+
+    def convert_inputs_to_api_kwargs(
+        self,
+        input: Optional[Any] = None,
+        model_kwargs: Dict = {},
+        model_type: ModelType = ModelType.UNDEFINED,
+    ) -> Dict:
+        """Convert the input and model_kwargs to api_kwargs for the LM Studio API."""
+        final_model_kwargs = model_kwargs.copy()
+        if model_type == ModelType.EMBEDDER:
+            if isinstance(input, str):
+                input = [input]
+            assert isinstance(input, Sequence), "input must be a sequence of text"
+            final_model_kwargs["input"] = input
+        elif model_type == ModelType.LLM:
+            messages = []
+            if input is not None and input != "":
+                messages.append({"role": "system", "content": "You are a helpful assistant. Provide a direct and concise answer to the user's question. Do not include any URLs or references in your response."})
```

Comment:
As we use uni-prompt where both system and user prompt are in the same jinja2 syntax, we only need one message here, and in default, we use role system.

```
messages.append({"role":"system", "content": input})
```

Please modify it to this.